### PR TITLE
Added a fix for symfony 5.3 container.

### LINKED ISSUE #592: Regression in getContainer with symfony 5.3
After the fix #584 there is no possibility to test `runCommand` with `$reuseKernel = true` when using symfony 5.3+.

### Preconditions
1. Use symfony 5.4 
2. Use `runCommand` test.

When there is a `$this->getContainer()` call before the `runCommand` it's not possible to reuse same kernel.
In the following code there is no possibility to reuse kernel created by `getContainer`.

```php
$client = $this->createMock(Client::class);

$this->getContainer()->get(Service::class)->setClient($client);
$this->runCommand('command', [], true);
```

### PR DESCRIPTION
**Pull request by biozshock**
Fixes #592

### CODE REVIEW

**Comment by biozshock** [comment]
No, that's the code needed to properly boot a kernel with the selected environment with static function `createKernel`

**Comment by alexislefebvre** [comment]
Why `$environment` was removed?

**Comment by biozshock** [comment]
it's not static.

**Comment by alexislefebvre** [comment]
Ok, thank you.

So we can't keep `$environment` because it can't be accessed in `createKernel()` ?

Please add a note in https://github.com/liip/LiipFunctionalTestBundle/blob/master/CHANGELOG.md to explain how to upgrade. It looks like it's a BC break so it may require a new major release.

**Comment by biozshock** [comment]
That's not a BC break, because you may still use old `environment` property without any consequences, except you'd get a silenced deprecation message.
Next major release should remove the deprecation and `environment` property.

File: src/Test/WebTestCase.php [hunk 1]
```diff
@@ -224,22 +226,60 @@ protected function getDependencyInjectionContainer(): ContainerInterface
...
             return $this->getDependencyInjectionContainer();
         }
 
         throw new \Exception("Method {$name} is not supported.");
     }
 
+    public function __set($name, $value)
+    {
+        if ($name !== 'environment') {
+            throw new \Exception(sprintf('There is no property with name "%s"', $name));
```
Comment (alexislefebvre) [inline src/Test/WebTestCase.php:253]: What if I extend WebTestCase and I use `$this->client` (or whatever new variable), will it throw an exception?

File: tests/Command/CommandTest.php [hunk 2]
```diff
@@ -85,8 +85,10 @@ public function testRunCommandWithoutOptionsAndNotReuseKernel(): void
         $this->assertIsBool($this->getDecorated());
         $this->assertTrue($this->getDecorated());
 
-        // Run command and not reuse kernel
-        $this->environment = 'prod';
```
Comment (alexislefebvre) [inline tests/Command/CommandTest.php:89]: Please duplicate this test to use `$this->environment = 'prod';` and not `self::$env = 'prod';`, to ensure BC.

File: src/Test/WebTestCase.php (hunk 1 above)
Comment (alexislefebvre) [reply src/Test/WebTestCase.php:253]: Actually some tests already declare new variables (e.g. [`$this->client =`](https://github.com/liip/LiipFunctionalTestBundle/pull/593/files#diff-2a2f8a611efd3fe4ee275790ac720835aab87fdcef0ba731d3754e40a3b92f01)) but I don't understand why it doesn't trigger this exception.

Comment (biozshock) [reply src/Test/WebTestCase.php:253]: This code is executed only when the variable does not exist. See https://www.php.net/manual/en/language.oop5.overloading.php#object.set

**Comment by alexislefebvre** [comment]
Could you please fix the failing test? Do you need help?

File: tests/Command/CommandTest.php (hunk 2 above)
Comment (biozshock) [reply tests/Command/CommandTest.php:89]: done

**Comment by alexislefebvre** [comment]
Thank you! These changes are available in the last release: https://github.com/liip/LiipFunctionalTestBundle/releases/tag/4.5.0

**Comment by mikysan** [comment]
> That's not a BC break, because you may still use old environment property without any consequences, except you'd get a silenced deprecation message.
> Next major release should remove the deprecation and environment property.

This is actually a BC break, in my company test suite the `WebTestCase` class was extended and `$environment` protected property overwritten, updating this library broke all the functional tests because the environment was once again defaulted to "test" (without us knowing it). Also the tests errors were about missing configuration without the environment explicitly showed by phpunit and no deprecation warning raised.

Also note that overriding `protected static $env = 'test';` in the extend class  doesn't fix the issue due to the use of `self::$env`  instead of `static::$env`. So there are actually two issues...

I didn't manage to find any issue about this, should I open them? If there's one already opened could you please link it?

**Comment by alexislefebvre** [comment]
Thanks for the report and sorry for the inconvenience.

There's no issue about this regression yet, feel free to add an issue and/or open a PR.

We could restore `protected $environment = 'test';` and a method to read `self::$env`, or `$this->environment` if the former has the default value?

**Comment by mikysan** [comment]
Ok, I'll open 2 issues and details there. Btw I won't be able to contribute with any PR until Friday hope it's ok.

**Pull request merged**